solution/backend/resources/v3serializers/resources.py

```diff
@@ -141,28 +143,73 @@ def validate_doc_type(self, value):
         return value
 
     def update(self, instance, validated_data):
-        # set basic fields
-        instance.url = validated_data.get('url', instance.url)
-        instance.description = validated_data.get('description', instance.description)
-        instance.name = validated_data.get('name', instance.name)
-        instance.docket_numbers = validated_data.get('docket_numbers', instance.docket_numbers)
-        instance.document_number = validated_data.get('document_number', instance.document_number)
-        instance.date = validated_data.get('date', instance.date)
-        instance.approved = validated_data.get('approved', instance.approved)
-        instance.doc_type = validated_data.get('doc_type', instance.doc_type)
-        instance.category = self.get_category()
+        created = self.context.get("created", False)
+
+        # set basic fields if instance is new
+        if created:
+            instance.url = validated_data.get('url', instance.url)
+            instance.description = validated_data.get('description', instance.description)
+            instance.name = validated_data.get('name', instance.name)
+            instance.docket_numbers = validated_data.get('docket_numbers', instance.docket_numbers)
+            instance.document_number = validated_data.get('document_number', instance.document_number)
+            instance.date = validated_data.get('date', instance.date)
+            instance.approved = validated_data.get('approved', instance.approved)
+            instance.doc_type = validated_data.get('doc_type', instance.doc_type)
+            instance.category = self.get_category()
 
         # set the locations on the instance
+        self.set_locations(instance, validated_data["sections"], validated_data["section_ranges"])
+
+        # reapply changelog if instance is not new
+        if not created:
+            self.apply_changelog(instance)
```

Comment:
I guess its explicit but couldnt if else do the same thing as the two ifs?

Reply:
Yes. In this case, as you said, it's just to be explicit and show the step-by-step nature of creating/updating an FR doc. But technically, setting the group could be moved to be within the first `if created` check.